src/wallet/wallet.cpp

```diff
@@ -4311,6 +4311,49 @@ bool CWallet::BackupWallet(const std::string& strDest)
     return database->Backup(strDest);
 }
 
+bool CWallet::PendingNameFirstUpdateExists(const std::string &name)
+{
+    LOCK(cs_wallet);
+    return namePendingMap.end() != namePendingMap.find(name);
+}
+
+bool CWallet::WritePendingNameFirstUpdate(const std::string &name, const valtype &tx, const uint256 &triggerTxid, const valtype &triggerName, const int32_t &triggerDepth)
+{
+    LOCK(cs_wallet);
+    CNamePendingData nameData;
+    nameData.setTx(tx);
+    nameData.setTriggerTxid(triggerTxid);
+    nameData.setTriggerName(triggerName);
+    nameData.setTriggerDepth(triggerDepth);
+
+    bool success = WalletBatch(*database).WriteNameFirstUpdate(name, nameData);
+    if(success)
+        namePendingMap[name] = nameData;
+
+    return success;
+}
+
+bool CWallet::ErasePendingNameFirstUpdate(const std::string &name)
+{
+    LOCK(cs_wallet);
+    bool success = WalletBatch(*database).EraseNameFirstUpdate(name);
+    if(success)
+        namePendingMap.erase(name);
+    return success;
+}
+
+bool CWallet::GetPendingNameFirstUpdate(const std::string &name, CNamePendingData *data)
+{
+    LOCK(cs_wallet);
+    std::map<std::string, CNamePendingData> pn = namePendingMap;
+    std::map<std::string, CNamePendingData>::iterator it = pn.find(name);
```

Comment:
Please use `auto` here to avoid the long type.

Reply:
Fixed.